db/1.9/modules/ROOT/pages/database-connector-select.adoc

```diff
@@ -0,0 +1,184 @@
+
+Use streaming with queries that return many records, such as in integration use cases. In Mule 4, streaming is transparent and always enabled.
+
+For example, if you submit a query that returns 10K rows by attempting to fetch all those rows at once results in both performance degradation, due to the big pull from the network, and the risk of running out of memory, because all the information must be loaded into RAM.
+
+With streaming, Database Connector fetches and processes only part of the query at one time, reducing the load on the network and memory.  This means that the connector does not fetch the 10K rows at once; instead, it fetches a smaller chunk, and once that chunk is consumed, it fetches the rest.
+
+You can also use the new repeatable streams mechanism where DataWeave and other components process the same stream many times, even in parallel.
+
+== Configure the Fetch Size and Max Rows Fields
+
+Because Mule runtime engine (Mule) enables Database Connector to manage streaming that does not mean that it's a good idea to move large chunks of data from the database to Mule. Even with streaming, a simple SQL query can return many rows, each one containing a lot of information. To limit the results, you can configure the *Fetch size* and *Max rows* fields.
+
+In the following example, you configure these fields for the Select operation. The syntax instructs Database Connector to fetch no more than 1000 rows (*Max rows* value), no more than 200 rows at a time (*Fetch size* value), significantly reducing network and memory load. The *Fetch size* value is enforced differently by different JDBC driver providers and often defaults to `10`.
+The combination limits the total amount of information that is retrieved (*Max rows* value) and guarantees that the data is returned from the database over the network in smaller chunks (*Fetch size* value):
+
+. In your Studio flow, select the *Select* operation.
+. In the operation configuration screen, set the *SQL Query Text* field to `select * from some_table`.
+. Set the *Advanced* tab, set the *Fetch size* field to `200`, and the *Max rows* field to `1000`.
+
+The following screenshot shows the configuration:
+
+.Fetch size and Max row configuration
+image::database-select-operation-4.png[.Fetch size and Max row configuration in Studio]
+
+In the XML editor, the `fetchSize` and `maxRows` configuration looks like this:
+
+[source,xml,linenums]
+----
+<db:select fetchSize="200" maxRows="1000" config-ref="dbConfig">
+  <db:sql>select * from some_table</db:sql>
+</db:select>
+----
+
+
+== Configure the Query Timeout Fields
+
+The following factors often cause delays in query execution:
+
+* An inefficient query, such as one having improper indexing that iterates over many rows
+* A busy RDBMS or network
+* A lock contention
+
+To avoid timeouts when executing queries configure the *Query timeout* and *Query time unit* fields. All Database Connector operations support setting a timeout.
```

Comment:
comma
```suggestion
To avoid timeouts when executing queries, configure the *Query timeout* and *Query time unit* fields. All Database Connector operations support setting a timeout.
```